spec/features/staff/users_can_sign_in_spec.rb

```diff
@@ -42,34 +42,140 @@ def log_in_via_form(user, remember_me: false)
+        # Given a user with 2FA enabled and a confirmed mobile number exists
+        user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: DateTime.now)
+
+        # When I log in with that user's email and password
+        visit root_path
+        log_in_via_form(user)
+
+        # Then there should be no link to check my mobile number
+        expect(page).not_to have_link "Check your mobile number is correct"
+
+        otp_at_time_of_login = user.current_otp
+        # And I should receive an OTP to my mobile number
+        expect(notify_client).to have_received(:send_sms).with({
+          phone_number: user.mobile_number,
+          template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
+          personalisation: {otp: otp_at_time_of_login}
+        })
+
+        # When I enter the one-time password before it expires
+        travel 3.minutes do
+          fill_in "Please enter your six-digit verification code", with: otp_at_time_of_login
+          click_on "Continue"
+        end
+
+        # Then I should be logged in
+        expect(page).to have_link(t("header.link.sign_out"))
+        expect(page).to have_content("Signed in successfully.")
+
+        # And I should be at the home page
+        expect(page).to have_content("You can search by RODA, Delivery Partner, or BEIS identifier, or by the activity's title")
+      end
 
-      # When I log in with that user's email and password
-      visit root_path
-      log_in_via_form(user)
+      scenario "unsuccessful OTP attempt" do
+        # Given a user with 2FA enabled and a confirmed mobile number exists
+        user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: DateTime.now)
+
+        # When I log in with that user's email and password
+        visit root_path
+        log_in_via_form(user)
 
-      otp_at_time_of_login = user.current_otp
-      # Then I should receive an OTP to my mobile number
-      expect(notify_client).to have_received(:send_sms).with({
-        phone_number: user.mobile_number,
-        template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
-        personalisation: {otp: otp_at_time_of_login}
-      })
-
-      # When I enter the one-time password before it expires
-      travel 3.minutes do
-        fill_in "Please enter your six-digit verification code", with: otp_at_time_of_login
+        # And I enter an incorrect OTP
+        fill_in "Please enter your six-digit verification code", with: "000000"
         click_on "Continue"
+
+        # Then I should not be logged in
+        expect(page).to have_content("Invalid two-factor verification code")
       end
+    end
 
-      # Then I should be logged in.
-      expect(page).to have_link(t("header.link.sign_out"))
-      expect(page).to have_content("Signed in successfully.")
+    context "user fails to confirm their mobile number" do
+      scenario "a successful login happens but the mobile number has not been confirmed" do
```

Comment:
Clarifying what we mean by "successful login" here would be useful: I assume we don't mean "can use RODA", but "we have a valid username/password"

Reply:
You're right, will fix